pkg/client/client.go

```diff
@@ -187,7 +203,20 @@ func getExecCredentialWithToken(token, expirationTimestamp string) (*datamodel.E
 	}, nil
 }
 
-func loadExecCredential() (*datamodel.ExecCredential, error) {
+type ClientHelpers interface {
+	LoadExecCredential() (*datamodel.ExecCredential, error)
+	GetServerURL(execCredential *datamodel.ExecCredential) (string, error)
+	GetTLSConfig(pemCAs []byte, nextProto string, insecureSkipVerify bool) (*tls.Config, error)
+	LoadAzureJSON() (*datamodel.AzureConfig, error)
```

Comment:
we should be able to mock I/O-related funcs using the pattern we discussed offline